string start
string[] stops
string end
---
string[] path
string stop